java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/instance/DeployedSystemTestResources.java

```diff
@@ -39,34 +43,40 @@ public DeployedSystemTestResources(SystemTestParameters parameters, SystemTestDe
         this.driver = driver;
     }
 
-    public void updateProperties(Consumer<SystemTestStandaloneProperties> config) {
-        config.accept(properties);
-        driver.saveProperties(properties);
-    }
-
     public SystemTestStandaloneProperties getProperties() {
         return properties;
     }
 
-    public void deployIfMissing() throws InterruptedException {
-        if (properties != null) {
+    public void deployIfMissing() throws InterruptedException, ExecutionException {
+        if (isDeployed()) {
             return;
         }
+        Future<?> future = executor.submit(() -> {
+            if (isDeployed()) {
+                return;
+            }
+            try {
+                deployIfMissingNoFailureTracking();
+            } catch (RuntimeException | InterruptedException e) {
+                failure = new InstanceDidNotDeployException(parameters.getSystemTestShortId(), e);
+                throw failure;
+            }
+        });
+        future.get();
+    }
+
+    private boolean isDeployed() {
         if (failure != null) {
             throw failure;
         }
-        try {
-            deployIfMissingNoFailureTracking();
-        } catch (RuntimeException | InterruptedException e) {
-            failure = new InstanceDidNotDeployException(parameters.getSystemTestShortId(), e);
-            throw e;
-        }
+        return properties != null;
     }
 
     public void resetProperties() {
```

Comment:
I think this method can be private. It doesn't look like it's called anywhere now though. It can be called at the end of deployIfMissingNoFailureTracking, or just after the properties are loaded.